lib/BSON/Double.pm

```diff
@@ -50,9 +50,23 @@ my $illegal = $^O eq 'MSWin32' && $] lt "5.022" ? qr/^$win32_specials/ : qr/^$un
 
 sub TO_JSON {
     my $copy = "$_[0]->{value}"; # avoid changing value to PVNV
-    return $_[0]->{value}/1.0 unless $copy =~ $illegal;
 
-    croak( "The value '$copy' is illegal in JSON" );
+    return { '$numberDouble' => 'Infinity' }
+        if $copy eq 'Inf';
```

Comment:
This isn't portable here and below.  You'll get text that matches `$win32_specials` on windows and it won't compare equal here.